lib/ConvoyLeader/src/App.cpp

```diff
@@ -132,6 +249,32 @@ void App::loop()
  * Private Methods
  *****************************************************************************/
 
+void App::positionTopicCallback(const String& payload)
+{
+    StaticJsonDocument<JSON_DOC_DEFAULT_SIZE> jsonPayload;
+    DeserializationError                      error = deserializeJson(jsonPayload, payload.c_str());
+
+    if (error != DeserializationError::Ok)
+    {
+        LOG_ERROR("JSON Deserialization Error %d.", error);
+    }
+    else
+    {
+        JsonVariant xPos = jsonPayload["X"];
+        JsonVariant yPos = jsonPayload["Y"];
+
+        if ((false == xPos.isNull()) && (false == yPos.isNull()))
+        {
+            LOG_DEBUG("Received position setpoint: x: %d y: %d", xPos.as<int32_t>(), yPos.as<int32_t>());

```

Comment:
Instead of converting it several times by using as<int32_t>(), convert them to int32_t variables and use them here and in the line below.